efopen/ef_aws_resolver.py

```diff
@@ -513,6 +513,103 @@ def cloudfront_origin_access_identity_oai_canonical_user_id(self, lookup, defaul
+    """
+    Args:
+        lookup: Cognito Federated Identity name, proto0-cms-identity-pool
+        default: the optional value to return if lookup failed; returns None if not set
+
+    Returns:
+        the constructed ARN for the cognito identity pool, else default/None
+    """
+    identity_pool_id = self.cognito_identity_identity_pool_id(lookup, default)
+
+    if not identity_pool_id:
+      return default
+
+    # The ARN has to be constructed because there is no boto3 call that returns the full ARN for a cognito identity pool
+    return "arn:aws:cognito-identity:${{AWS::Region}}:${{AWS::AccountId}}:identitypool/{}".format(identity_pool_id)
+
+  def cognito_identity_identity_pool_id(self, lookup, default=None):
+    """
+    Args:
+        lookup: Cognito Federated Identity name, proto0-cms-identity-pool
+        default: the optional value to return if lookup failed; returns None if not set
+
+    Returns:
+        the Cognito Identity Pool ID corresponding to the given lookup, else default/None
+    """
+    list_limit = 60
+    response = EFAwsResolver.__CLIENTS["cognito-identity"].list_identity_pools(MaxResults=list_limit)
+
+    while True:
+      if not response.has_key("IdentityPools"):
+        break
+
+      # Loop through all the identity pools
+      for pool in response["IdentityPools"]:
+        if pool["IdentityPoolName"] == lookup:
+          return pool["IdentityPoolId"]
+
+      # No match found on this page, but there are more pages
+      if response.has_key("NextToken"):
+        response = EFAwsResolver.__CLIENTS["cognito-identity"].list_identity_pools(MaxResults=list_limit,
+                                                                                   NextToken=response["NextToken"])
+      else:
+        break
+
+    return default
+
+  def cognito_idp_user_pool_arn(self, lookup, default=None):
+    """
+    Args:
+        lookup: Cognito User Pool name, proto0-cms-user-pool
+        default: the optional value to return if lookup failed; returns None if not set
+
+    Returns:
+        the User Pool ARN corresponding to the given lookup, else default/None
+    """
+    user_pool_id = self.cognito_idp_user_pool_id(lookup, default)
+    if not user_pool_id:
+      return default
+
+    list_limit = 60
+    response = EFAwsResolver.__CLIENTS["cognito-idp"].describe_user_pool(UserPoolId=user_pool_id)
+
+    if not response.has_key("UserPool"):
+      return default
+
+    return response["UserPool"]["Arn"]
+
+  def cognito_idp_user_pool_id(self, lookup, default=None):
+    """
+    Args:
+        lookup: Cognito User Pool name, proto0-cms-user-pool
+        default: the optional value to return if lookup failed; returns None if not set
+
+    Returns:
+        the User Pool ID corresponding to the given lookup, else default/None
+    """
+    list_limit = 60
+    response = EFAwsResolver.__CLIENTS["cognito-idp"].list_user_pools(MaxResults=list_limit)
+    while True:
```

Comment:
This can be rewritten as:
```suggestion
    while "UserPools" in response:
```

Reply:
I usually get angry in code reviews. This.... I like.